codec-smtp/src/main/java/io/netty/handler/codec/smtp/LastSmtpContent.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.smtp;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+
+/**
+ * The last part of a sequence of {@link SmtpContent}s that are send after a {@code DATA} request.
+ * Be aware that a {@link SmtpContent} / {@link LastSmtpContent} sequence must always use CRLF as line delimiter
+ * and that lines that start with a DOT must be escapted via an extra DOT as
+ * specified by <a href="https://www.ietf.org/rfc/rfc2821.txt">RFC2821</a>.
+ */
+public interface LastSmtpContent extends SmtpContent {
+
+    /**
+     * Empty {@link LastSmtpContent}.
+     */
+    LastSmtpContent EMPTY_LAST_CONTENT = new LastSmtpContent() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 51 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1188)